go/libraries/doltcore/schema/tag.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"math"
+	"math/rand"
+	"time"
+)
+
+const (
+	// TODO: increase ReservedTagMin to 1 << 63 once numeric marshalling is fixed
+	// ReservedTagMin is the start of a range of tags which the user should not be able to use in their schemas.
+	ReservedTagMin uint64 = 1 << 50
+
+	//
+	SystemTableReservedMin uint64 = 1 << 51
+
+	// InvalidTag is used as an invalid tag
+	InvalidTag uint64 = math.MaxUint64
+)
+
+const (
+	// Tags for dolt_docs table
+	DocNameTag = iota + SystemTableReservedMin
```

Comment:
This is one place where this style of const declaration is a bad idea. These constants are persisted to repos. That means if I reorder this list, it silently changes the values and introduces a breaking change to dolt.

Each table should have its own const block with its own predefined keyspace. We have lots of overhead here. 1000 tags per system table gives us plenty of room to grow.

Also add a comment warning against reordering the consts